mlrun/db/httpdb.py

```diff
@@ -1111,6 +1111,60 @@ def create_project(
         )
         return mlrun.projects.MlrunProject.from_dict(response.json())
 
+    def create_project_secrets(
+        self,
+        project: str,
+        provider: Union[
+            str, schemas.SecretProviderName
+        ] = schemas.SecretProviderName.vault,
+        secrets: dict = None,
+    ):
+        if isinstance(provider, schemas.SecretProviderName):
+            provider = provider.value
+        path = f"projects/{project}/secrets"
+        body = {"provider": provider, "secrets": secrets}
+        error_message = f"Failed creating secret provider {project}/{provider}"
+        self.api_call(
+            "POST", path, error_message, body=dict_to_json(body),
+        )
+
+    def get_project_secrets(
+        self,
+        project: str,
+        token: str,
+        provider: Union[
+            str, schemas.SecretProviderName
+        ] = schemas.SecretProviderName.vault,
+        secrets: List[str] = None,
+    ) -> dict:
+        if isinstance(provider, schemas.SecretProviderName):
+            provider = provider.value
+        path = f"projects/{project}/secrets"
+        params = {"provider": provider, "secret": secrets}
+        headers = {schemas.HeaderNames.secret_store_token: token}
+        error_message = f"Failed retrieving secrets {project}/{provider}"
+        result = self.api_call(
+            "GET", path, error_message, params=params, headers=headers,
+        )
+        return result.json()["secrets"]
```

Comment:
This goes back to the same discussion we've been having again and again. I don't think we agreed on one way or the other...

Reply:
The decision was that something with some kind of schema (and therefore auto-completion and stuff) is preferred.
If there are both API schema and SDK schema, the SDK is preferred.
For feature-sets, we left it a dict since it will be changed to the SDK schema soon (when it will be added)
For this one there won't be any SDK schema (not soon at least), so API schema is preferred over plain dict

Reply:
ok. Since this method is called by `MlrunProject.get_vault_secrets()`, I'm going to return the schema here and have the SDK API return just the `.secrets` part of it to the caller. This will have the ease-of-use for the SDK while maintaining the schema.